internal/cmd/test.go

```diff
@@ -135,3 +137,56 @@ func runCMD(mainFunc func(), args []string) (err error) {
 
 	return err
 }
+
+type CommonCMDTestCases struct {
+	ExpectedExitCode int
+	ExpectedStdout   string
+	ExpectedStderr   string
+}
+
+func RunCommonCMDTests(test *testing.T, mainFunc func(), args []string, commonCases CommonCMDTestCases, prefix string) (string, string, int) {
+	stdout, stderr, exitCode, err := RunCMDTest(test, mainFunc, args)
+	if err != nil {
+		test.Errorf("%sCMD run returned an error: '%v'.", prefix, err)
+	}
+
+	if commonCases.ExpectedStderr != "" {
```

Comment:
Why is this check necessary?
If we always just do the normal equality check, then we also don't need to length check below.
stderr will either be exactly as we want (which may be empty) or we fail.